torch/csrc/jit/register_quantized_ops.cpp

```diff
@@ -28,7 +28,10 @@ using caffe2::int8::Int8TensorCPU;
 namespace {
 
 caffe2::Tensor from_at_tensor(const c10::IValue& v) {
-  return caffe2::Tensor(autograd::Variable(std::move(v).toTensor()).tensor_data());
+  // We use `._tensor_data_deprecated()` here, because Caffe2 can't handle Variables yet.
+  // After the Variable/Tensor merge, Caffe2 will be able to handle non-requires-grad Variables,
+  // and we will remove `._tensor_data_deprecated()` here.
+  return caffe2::Tensor(autograd::Variable(std::move(v).toTensor())._tensor_data_deprecated());
```

Comment:
how is Caffe2 going to work, though?  Presumably we need a NoGradMode / NonVariableTypeMode somewhere before we call into it, and that can be done before we merge variable and tensor?

Reply:
This is being addressed in https://github.com/pytorch/pytorch/pull/22473.